src/ReactPlaceholder.tsx

```diff
@@ -1,8 +1,39 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
 import * as placeholders from './placeholders';
 
-export default class ReactPlaceholder extends React.Component {
+export type CommonProps = {
+  children: React.ReactNode,
+  /** pass `true` when the content is ready and `false` when it's loading */
+  ready: boolean,
+  /** delay in millis to wait when passing from ready to NOT ready */
+  delay?: number,
+  /** if true, the placeholder will never be rendered again once ready becomes true, even if it becomes false again */
+  firstLaunchOnly?: boolean,
+  className?: string,
+  style?: React.CSSProperties
+}
+
+export type Props = (CommonProps & {
+  /** type of placeholder to use */
+  type: 'text' | 'media' | 'textRow' | 'rect' | 'round',
+  /** number of rows displayed in 'media' and 'text' placeholders */
+  rows?: number,
+  /** color of the placeholder */
+  color?: string,
+  /** pass true to show a nice loading animation on the placeholder */
+  showLoadingAnimation?: boolean,
+  customPlaceholder?: undefined
+}) | (CommonProps & {
+  /** pass any renderable content to be used as placeholder instead of the built-in ones */
+  customPlaceholder?: React.ReactNode | React.ReactElement<{ [k: string]: any }>,
+  type?: undefined,
+  rows?: undefined,
+  color?: undefined,
+  showLoadingAnimation?: undefined
+})
```

Comment:
You could use https://github.com/gcanti/prop-types-ts and avoid duplication, what do you think?